### PR TITLE
Add Reporting Vulnerabilities section to Security Policy

### LINKED ISSUE #16919: [Security] [Doc] no advice on how to report vulnerabilities
**Describe the bug**
security: no advice on how to report vulnerabilities

**To Reproduce**
1. visit: https://github.com/apache/pulsar/issues/new/choose
2. click button "view policy": https://github.com/apache/pulsar/security/policy
3. follow link to: https://pulsar.apache.org/docs/security-policy-and-supported-versions/

see screenshot

**Expected behavior**
an advice how to report vulnerabilities - or is it fine to just open an issue?

**Screenshots**
2022-08-02_15h43_40

### PR DESCRIPTION
**Pull request by tisonkun**
Fixes #16919

- [x] `doc`

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @hpvd @sijie @merlimat @eolivelli

**Review by codelipenghui — APPROVED** [review]

File: site2/docs/security-policy-and-supported-versions.md [hunk 1]
```diff
@@ -14,6 +14,12 @@ https://pulsar.apache.org/docs/en/security-overview/.
 The Pulsar community will announce security vulnerabilities and how to mitigate them on the [users@pulsar.apache.org](mailto:users@pulsar.apache.org).
 For instructions on how to subscribe, please see https://pulsar.apache.org/contact/.
 
+## Reporting Vulnerabilities
+
+The Pulsar community follows the ASF [vulnerability handling process](https://apache.org/security/#vulnerability-handling).
+
+To report a new vulnerability you have discovered please follow the [ASF vulnerability reporting process](https://apache.org/security/#reporting-a-vulnerability). Besides, you can send one plain-text email for each vulnerability to [private@pulsar.apache.org](mailto:private@pulsar.apache.org).
```
Comment (Anonymitaet) [inline site2/docs/security-policy-and-supported-versions.md:21]: Are users allowed to send emails to `security@apache.org` as well?

Comment (tisonkun) [reply site2/docs/security-policy-and-supported-versions.md:21]: Yes. See also the links in the content:

* https://www.apache.org/security/projects.html
* https://apache.org/security/#vulnerability-handling
* https://apache.org/security/#reporting-a-vulnerability

Comment (Anonymitaet) [reply site2/docs/security-policy-and-supported-versions.md:21]: If users are allowed / encouraged to send emails to `security@apache.org` and `private@pulsar.apache.org`

1. Why only add `private@pulsar.apache.org` to guide (in this PR)?

2. When reporting a Pulsar security issue, is there any difference between `security@apache.org` and `private@pulsar.apache.org`? Or user can report to any of them, they are the same?

Comment (tisonkun) [reply site2/docs/security-policy-and-supported-versions.md:21]: > Why only add private@pulsar.apache.org to guide (in this PR)?

It starts with `Besides, ...`. Actually, we can remove the possibility to report to private@ to make it easy. A vulnerability reported to security@ will be carried to the PMC (private@) if necessary.

Comment (tisonkun) [reply site2/docs/security-policy-and-supported-versions.md:21]:
```suggestion
To report a new vulnerability you have discovered please follow the [ASF vulnerability reporting process](https://apache.org/security/#reporting-a-vulnerability).
```

... like this.

Comment (tisonkun) [reply site2/docs/security-policy-and-supported-versions.md]: Updated.

**Review by Anonymitaet — APPROVED** [review]

**Comment by lhotari** [comment]
I hope that the information about the vulnerability handling process could also be added to SECURITY.md . Duplication of information is justified in this case. Hiding relevant information behind yet another click could result in information being missed.

I'd suggest that we also add these sentences to SECURITY.md:

> The Pulsar community follows the ASF [vulnerability handling process](https://apache.org/security/#vulnerability-handling).
> 
> To report a new vulnerability you have discovered please follow the [ASF vulnerability reporting process](https://apache.org/security/#reporting-a-vulnerability).

**Comment by lhotari** [comment]
Oh, I see that @tisonkun already suggested inlining the information in  https://github.com/apache/pulsar/issues/16919#issuecomment-1207568465 .

**Comment by tisonkun** [comment]
@lhotari I'm OK to duplicate the information but the current state is acceptable for me and I think it resolves the original issue.

If you'd like to duplicate the information, you can submit a patch :)

**Comment by lhotari** [comment]
> @lhotari I'm OK to duplicate the information but the current state is acceptable for me and I think it resolves the original issue.
> 
> If you'd like to duplicate the information, you can submit a patch :)

@tisonkun I made #17039 as a follow-up

**Pull request merged**